tests/unit/utils/cache-test.js

```diff
@@ -0,0 +1,154 @@
+import { module, test } from 'qunit';
+import { QueryCache } from 'ember-scoped-query-cache/utils';
+
+module('Unit | Utility | query-queryCache');
+
+test('Creates queryCache object with internal map', function (assert) {
+  let queryCache = new QueryCache();
+
+  assert.ok(queryCache._internalCache);
+});
+
+test('QueryCache#add skips adding if key is undefined', function(assert) {
+  let queryCache = new QueryCache();
+
+  queryCache.add('my-type');
+
+  assert.notOk(queryCache._internalCache['my-type']);
+});
+
+test('QueryCache#add skips adding if value is undefined', function(assert) {
+  let queryCache = new QueryCache();
+
+  queryCache.add('my-type', 'mykey');
+
+  assert.notOk(queryCache._internalCache['my-type']);
+});
+
+test('QueryCache#add adds item to queryCache when all params specified', function(assert) {
+  let queryCache = new QueryCache();
+
+  queryCache.add('my-type', 'mykey', 'myvalue');
+
+  assert.ok(queryCache._internalCache['my-type']);
```

Comment:
Yes creating the `has` API affords both better testing, but may also be handy when there is more fine grained caching.